Mathlib/RingTheory/Ideal/Prod.lean

```diff
@@ -199,3 +199,13 @@ theorem ideal_prod_prime (I : Ideal (R × S)) :
     · exact isPrime_ideal_prod_top'
 
 end Ideal
+
+instance [IsPrincipalIdealRing R] [IsPrincipalIdealRing S] : IsPrincipalIdealRing (R × S) where
+  principal I := by
+    rw [I.ideal_prod_eq]
+    obtain ⟨r, hr⟩ := Submodule.IsPrincipal.principal (I.map (RingHom.fst R S))
+    obtain ⟨s, hs⟩ := Submodule.IsPrincipal.principal (I.map (RingHom.snd R S))
+    simp_rw [hr, hs, Ideal.submodule_span_eq]
```

Comment:
I'm surprised we don't have a version of this result specialized to `Ideal`, which already rewrites with `Ideal.submodule_span_eq`. But apparently we don't!

Reply:
If I move it to the other file I can golf it to
```
instance [IsPrincipalIdealRing R] [IsPrincipalIdealRing S] : IsPrincipalIdealRing (R × S) where
  principal I := by
    rw [I.ideal_prod_eq, ← (I.map _).span_singleton_generator,
      ← (I.map (RingHom.snd R S)).span_singleton_generator,
      ← Ideal.span_prod (iff_of_true (by simp) (by simp)), Set.singleton_prod_singleton]
    exact ⟨_, rfl⟩
```
Should I do that? The transitive imports are only 968 compared to 931.

Reply:
Yes, let's go for neatness of proof over import count here.